pyk/src/pyk/kore/syntax.py

```diff
@@ -1651,8 +1651,7 @@ def let(self, *, app: App | None = None) -> LeftAssoc:
         return LeftAssoc(app=app)
 
     def let_patterns(self, patterns: Iterable[Pattern]) -> LeftAssoc:
-        () = patterns
-        return self
+        return self.let(app=self.app.let(args=patterns))
```

Comment:
This could instead be
```
(app,) = patterns
assert isinstance(app, App)
return self.let(app=app)
```
Which would also catch the entire `App` during `top_down`/`bottom_up`. @tothtamas28 any opinions? Or, alternatively (see my other comment)

Reply:
I'm not perfectly happy with either of these solutions:
1. The one solution does not operate on the direct child.
2. The other solution is not type safe (statically).

But then the current solution is not ideal either. I think (2) is more straightforward from the programmer's perspective, so I lean towards that.

If we do this change however:
- There's some code that could be simplified, it would be good to take care of that as well.
- It's a breaking change, so it needs to be tested on `evm-semantics` and `kontrol`.

Would it be acceptable for you to work around this for now (the same way e.g. `kore_to_kast` does)? Then I'll open a PR with the change and the simplifications, and do the downstream testing.

Reply:
Yeah, I found a workaround that I can use.